#! /usr/bin/env python
PACKAGE='hockey_shot'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Lidar Filter Stuff
#lidar_filter = gen.add_group("Lidar Filter", type="tab")

#       Name                        Type       Lvl    Description                                           Default     Min      Max
gen.add("x_min",                    double_t,  0,     "Region of interest",                                 -1.9,       -10.0,   0.0)
gen.add("x_max",                    double_t,  0,     "Region of interest",                                 0.67,       0.0,     10.0)
gen.add("y_min",                    double_t,  0,     "Region of interest",                                 -1.4,       -10.0,   0.0)
gen.add("y_max",                    double_t,  0,     "Region of interest",                                 0.33,       0.0,     10.0)
gen.add("z_min",                    double_t,  0,     "Region of interest",                                 -1.0,       -10.0,   2.0)
gen.add("z_max",                    double_t,  0,     "Region of interest",                                 4.0,        2.0,     10.0)
gen.add("uniform_thresh",           double_t,  0,     "Uniform sample thresh",                              0.1,        0.001,   1.0)
gen.add("cluster_tol",              double_t,  0,     "Distance to include point in cluster",               0.02,         0.001,   2.0)
gen.add("min_cluster_size",         int_t,     0,     "Minimum allowed number of points in a cluster",      50,          1,       500)
gen.add("max_cluster_size",         int_t,     0,     "Maximum allowed number of points in a cluster",      5000,         1,       1000)

gen.add("distance_thresh",              double_t,  0,     "Distance to include point in cluster",           5,         0.001,   10.0)
gen.add("pcolor_thresh",              double_t,  0,     "Distance to include point in cluster",             6,         0.001,   20.0)
gen.add("rcolor_thresh",              double_t,  0,     "Distance to include point in cluster",             5,         0.001,   20.0)
gen.add("min_cluster",              double_t,  0,     "Distance to include point in cluster",               100,         0.001,  500.0)

gen.add("num_normal_neighbors",     int_t,     0,     "Number of neighbors to search to estimate normals",  50,         5,       100)

gen.add("red",    int_t,     0,     "Max Red in colored clourd",       100,     0,     255)
gen.add("blue",    int_t,     0,     "Max Blue in colored clourd",       255,     0,     255)
gen.add("green",    int_t,     0,     "Max Green in colored clourd",       255,     0,     255)
gen.add("point_distance",   double_t,  0,     "Distance for point in pocket",    3.5,         2,  4)

exit(gen.generate(PACKAGE, PACKAGE, "HockeyShot"))
